src/crypto/shelley/metadataUtils.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable camelcase */
+// // @flow
+import {
+  TransactionMetadatum,
+  encode_json_str_to_metadatum,
+  decode_metadatum_to_json_str,
+  MetadataJsonSchema,
+  GeneralTransactionMetadata,
+  BigNum,
+  TransactionMetadata,
+} from '@emurgo/react-native-haskell-shelley'
+
+export type JSONMetadata = {|
+  label: string,
+  data: {...},
+|}
+
+export function createMetadata(
+  metadata: Array<JSONMetadata>
+): TransactionMetadata {
+  const transactionMetadata = GeneralTransactionMetadata.new()
+
+  metadata.forEach((meta: JSONMetadata) => {
+    const metadatum = encode_json_str_to_metadatum(
```

Comment:
Also note that since `encode_json_str_to_metadatum()` is an async call, a `forEach` wouldn't work.